compiler/il/OMRNode_inlines.hpp

```diff
@@ -171,11 +171,20 @@ OMR::Node::getType()
    return self()->getDataType();
    }
 
-TR_NodeUseAliasSetInterface
+TR_UseOnlyAliasSetInterface
 OMR::Node::mayUse()
    {
-   TR_NodeUseAliasSetInterface aliasSetInterface(self());
-   return aliasSetInterface;
+   if (self()->getOpCode().isLikeUse() && self()->getOpCode().hasSymbolReference())
+      {
+      TR_UseOnlyAliasSetInterface aliasSetInterface(self()->getSymbolReference());
+      return aliasSetInterface;
+      }
+   else 
+      {
+      //if there is no symbolreference, then return an empty aliseset
+       TR_UseOnlyAliasSetInterface aliasSetInterface(NULL);
```

Comment:
Should this path ever get executed ? If yes, it may be worth checking if it's possible to check if a symbol reference exists on the caller side before calling it (in case there is only one place).

Reply:
This code should be only be executed if `self()->getOpCode().hasSymbolReference()` or `.isLikeUse()` evaluate to `false`, which is checked in the `if`/`else` expression.